backend/hitas/templates/components/improvements_table.jinja

```diff
@@ -0,0 +1,141 @@
+            </tr>
+        {% endfor %}
+        <tr style="border-top: 1px solid gray; padding-top: 2px;">
+            <td class="improvement-cell"><b>Parannukset yhteensä</b></td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td></td>
+            <td class="improvement-cell text-right"><b>{{ improvements.summary.value|intcomma }}</b></td>
+            <td class="improvement-cell text-right">*TODO*</td>
+            <td class="improvement-cell text-right"><b>{{ improvements.summary.depreciation|intcomma }}</b></td>
+            <td class="improvement-cell text-right"><b>{{ improvements.summary.value_for_apartment|intcomma }}</b></td>
+        </tr>
+    </table>
+
+    {% set excess = improvements.summary.excess %}
+    {% if excess %}
+        <p>Parannuksen omavastuu on {{ excess.surface_area|intcomma }} m² x {{ excess.value_per_square_meter|intcomma }} euroa/m² = {{ excess.total|intcomma }} euroa. Arvoa korottavaan osuuteen tehdään indeksitarkistus ilman poistoja.</p>
+    {% endif %}
+{% endmacro %}
+
+
+{% macro housing_company_improvements_table_post_2011(index_calculation) %}
+    <h3 class="bordered-header">HUONEISTON OSUUS YHTIÖN PARANNUKSISTA</h3>
+    <table class="improvements-table">
+        <tr style="border-bottom: 1px solid black;">
+            <td class="improvement-cell" style="width: 172px;"><b>Parannus</b></td>
+            <td class="improvement-cell improvement-column" style="width: 60px;"><b>Arvo</b></td>
+            <td class="improvement-cell improvement-column" style="width: 80px;"><b>Valmistunut</b></td>
+            <td class="improvement-cell improvement-column" style="width: 80px;"><b>Arvoa korot-<br>tava osuus</b></td>
+            <td class="improvement-cell improvement-column" style="width: 70px;"><b>Poistoaika</b></td>
+            <td class="improvement-cell improvement-column" style="width: 60px;"><b>Poiston määrä</b></td>
+            <td class="improvement-cell improvement-column" style="width: 80px;"><b>Hyväksytty yhtiö</b></td>
+            <td class="improvement-cell improvement-column" style="width: 80px;"><b>Hyväksytty huoneisto</b></td>
+        </tr>
+        {% set improvements = index_calculation.calculation_variables.housing_company_improvements %}
+        {% for improvement in improvements["items"] %}
+            <tr style="padding-top: 2px;">
+                <td class="improvement-cell">{{ improvement.name }}</td>
+                <td class="improvement-cell text-right">{{ improvement.value|intcomma }}</td>
+                <td class="improvement-cell text-right">{{ improvement.completion_date }}</td>
+                <td class="improvement-cell text-right">{{ improvement.value_added|intcomma }}</td>
+                <td class="improvement-cell text-right">TODO</td>
+                <td class="improvement-cell text-right">{{ improvement.depreciation }}</td>
```

Comment:
as these are `None` for apartments completed after 2011 you can omit this one and the one above :)

old system shows these as `0v 0kk` and `0` but as these are not used in calculations I think it's better just to omit those

Reply:
Removed deprecation-related columns :)